types/pnpapi/index.d.ts

```diff
@@ -1,41 +1,60 @@
-// Type definitions for pnpapi 0.0
+// Type definitions for pnpapi 3.0
 // Project: https://yarnpkg.github.io/berry/advanced/pnpapi
 // Definitions by: Maël Nison <https://github.com/arcanis>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
+/// <reference types="node"/>
 
-export interface PhysicalPackageLocator {
-    name: string;
-    reference: string;
+declare namespace NodeJS {
+    interface ProcessVersions {
+        pnp?: string;
+    }
 }
 
-export interface TopLevelPackageLocator {
-    name: null;
-    reference: null;
-}
+declare module 'pnpapi' {
```

Comment:
This is going to substantially worsen the end-user experience of this package. You can separate out the `NodeJS` augmentation into a different file and leave this file's format as-is.

Reply:
Sounds good, I'll check tomorrow. Why is the declare module approach discouraged btw? What use cases does it break?